src/main/java/com/microboxlabs/events/sse/emitter/EventStreamResource.java

```diff
@@ -48,7 +49,12 @@ public class EventStreamResource {
     })
     public Response emitEvent(
             @RequestBody(description = "The event data to be broadcast", required = true) EventData eventData) {
-        eventEmitter.sendAndForget(eventData);
+        if (eventEmitter.hasRequests()) {
+            eventEmitter.sendAndForget(eventData);
+        } else {
+            logger.warning("No subscribers for event: " + eventData);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
```

Comment:
Hi @ricardorodriguezgonzalez,

Thank you for your work on this PR. I appreciate the addition of the check for subscribers before emitting an event. However, I have a couple of suggestions regarding the status code used when there are no subscribers and the corresponding API documentation.

**1. Status Code Change:**
Currently, the code returns a 500 Internal Server Error when there are no subscribers. While this does indicate a problem, it's not the most appropriate status code for this situation. An Internal Server Error typically suggests a server-side issue that prevented the request from being fulfilled, which isn't quite what's happening here.

Instead, I'd recommend using 412 Precondition Failed. Here's why:
- 412 Precondition Failed indicates that access to the target resource has been denied due to not meeting a precondition that the server requires.
- In our case, having active subscribers can be considered a precondition for emitting an event.
- This status code more accurately represents the state of the system to the client, allowing for more appropriate error handling on their end.

Could you please update the code to use 412 instead? Here's an example of how it might look:


```java
@RequestBody(description = "The event data to be broadcast", required = true) EventData eventData) {
  if (eventEmitter.hasRequests()) {
  eventEmitter.sendAndForget(eventData);
  return Response.accepted().build();
  } else {
  logger.warning("No subscribers for event: " + eventData);
  return Response.status(Response.Status.PRECONDITION_FAILED)
  .entity("No subscribers available to receive the event")
  .build();
  }
}
```

**2. API Documentation Update:**
We also need to update the API documentation to reflect this new possible response. Please modify the `@APIResponse` annotation to include the 412 status code:

```java
@APIResponses(value = {
@APIResponse(responseCode = "202", description = "Event accepted for broadcasting"),
@APIResponse(responseCode = "400", description = "Invalid event data"),
@APIResponse(responseCode = "412", description = "No subscribers available to receive the event")
})
```


These changes will provide a more accurate representation of the API's behavior, improve its usability, and ensure our documentation is up-to-date.

Let me know if you have any questions or need any clarification on this feedback.

Thanks again for your contribution!

Reply:
Hi @ricardorodriguezgonzalez,

we've been discussing the best approach for handling the "no subscribers" scenario. I'd like to propose a different solution that I believe better fits our use case.

Instead of using `412 Precondition Failed`, let's go with `202 Accepted`. Here's the rationale:

1. The lack of subscribers is a server-side condition, not a client error. Using a `4xx` status code (like 412) would incorrectly imply that the client needs to correct something.

2. `202` Accepted indicates that the request has been accepted for processing, but the processing has not been completed. This aligns well with our scenario where we accept the event, even if we can't broadcast it immediately due to lack of subscribers.

3. This approach is more flexible, as it allows for the possibility that subscribers might connect later and could potentially receive the event.

Here's how the updated code might look:

```java
public Response emitEvent(
  @RequestBody(description = "The event data to be broadcast", required = true) EventData eventData) {
  if (eventEmitter.hasRequests()) {
  eventEmitter.sendAndForget(eventData);
  return Response.accepted().build();
  } else {
  logger.warning("No subscribers for event: " + eventData);
  return Response.accepted()
  .entity("Event accepted, but no subscribers are currently available to receive it")
  .build();
  }
}
```

And the updated API documentation:

```java
@APIResponses(value = {
@APIResponse(responseCode = "202", description = "Event accepted for processing. If subscribers are available, it will be broadcast immediately; otherwise, it may be queued or discarded based on the current implementation."),
@APIResponse(responseCode = "400", description = "Invalid event data")
})
```


This approach maintains a consistent `202` response while providing additional information in the response body when there are no subscribers. It also keeps our API behavior more predictable for clients.

Could you please update the PR with these changes? Let me know if you have any questions or need any clarification.